pkg/daemon/writer.go

```diff
@@ -43,7 +44,11 @@ func (nw *NodeWriter) Run(stop <-chan struct{}) {
 		case <-stop:
 			return
 		case msg := <-nw.writer:
-			msg.responseChannel <- setNodeAnnotations(msg.client, msg.node, msg.annos)
+			if msg.annos != nil {
+				msg.responseChannel <- setNodeAnnotations(msg.client, msg.node, msg.annos)
+			} else if msg.taint != nil {
```

Comment:
Minor: I'd just make this another `if` instead of `else if`. Not that the current code currently tries to set both annotations and taints at the same time, but one would expect it'd handle it correctly.